Fixing issues: round 1

raffazizzi opened the pull request.

raffazizzi committed:
Stopped passing collection to CategoryView classes; it's not needed.

Fixes #16

raffazizzi committed:
Renamed "name" to "catName"

When triggering events for ProjectsView, we need to pass info about the current category. This is done with an object with two properties: "catType" and "name". This commit renames "name" to the clearer "catName".

Fixes #15

raffazizzi committed:
Fixed count of projects in a given category

The count needed to be recursive in order to account for all projects in nested categories.

Fixes #14

raffazizzi committed:
Add image placeholder when project image is missing.

This commit also adds a temporary image that will eventually be hosted on the site. At that point the hardcoded URL to the image will need to be changed in the template.

Fixes #4

raffazizzi committed:
Created dedicated link target for showing only one category.

This new link replaces clicking on the category name. The link ("only") appears on mouse over or when tapping on a touch device.
Also added a temporary CSS file needed to set the element invisible at start.

Fixes #3

trevormunoz commented:
Need to resolve minor issue #20 to proceed with review

trevormunoz commented:
Question relating to #14:
Shouldn't the top-level count here be 74 not 68? (Not entirely sure because it's hard to tell exactly without styling.)

<img width="272" alt="screen shot 2015-12-01 at 11 04 07 am" src="https://cloud.githubusercontent.com/assets/351120/11506100/a3cdd442-981b-11e5-87d5-21fda98f563a.png">

trevormunoz committed:
Make underscore dependency explicit

Should fix #20

raffazizzi committed:
Fixed project count bug introduced with 3c3474a.

Now the recursive function correctly get "subset" as an attribute, not as a property.

Fixes #14

raffazizzi commented:
Yes I had introduced a bug. The count is now back to 84, as it should be (some projects are just "projects" and do not belong to a subgroup).
